src/containerapp/azext_containerapp/java_component_decorator.py

```diff
@@ -214,3 +216,126 @@ def process_loaded_scg_route(self):
                 raise ValidationError(f'The "filters" field must be a list in route {route["id"]}. Please see https://aka.ms/gateway-for-spring-routes-yaml for a valid Gateway for Spring routes YAML spec.')
 
         return yaml_scg_routes.get('springCloudGatewayRoutes')
+
+
+class JavaComponentCreateDecorator(BaseJavaComponentDecorator):
+
+    def create(self):
+        try:
+            return self.client.create(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                environment_name=self.get_argument_environment_name(), name=self.get_argument_java_component_name(),
+                java_component_envelope=self.java_component_def,
+                no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            stringErr = str(e)
+            if "JavaComponentsNotAllowedForSubscription" in stringErr:
+                raise CLIInternalError("Java Components operations are not allowed for the subscription, please use 'az feature register --namespace  Microsoft.App --name JavaComponentsPreview' to register this feature.")
+
+            handle_raw_exception(e)
+
+    # need handle already set configurations
+    def set_up_set_configurations(self):
+        if self.get_argument_set_configurations() is not None:
+            configuration_list = self.parse_configurations(self.get_argument_set_configurations())
+            self.java_component_def["properties"]["configurations"] = configuration_list
```

Comment:
Is it safe to replace the whole list specified by `--set-configurations`? Do we need to preserve the original values if not set?